ote_cli/ote_cli/utils/hpo.py

```diff
@@ -0,0 +1,605 @@
+"""
+Utils for HPO with hpopt
+"""
+
+import builtins
+import collections
+import importlib
+import copy
+import multiprocessing
+import os
+import pickle
+import shutil
+import subprocess
+import sys
+import time
+from os import path as osp
+from pathlib import Path
+from typing import Optional
+
+import numpy
+import torch
+import yaml
+from ote_sdk.configuration.helper import create
+from ote_sdk.entities.model import ModelEntity, ModelStatus
+from ote_sdk.entities.model_template import TaskType
+from ote_sdk.entities.subset import Subset
+from ote_sdk.entities.task_environment import TaskEnvironment
+from ote_sdk.entities.train_parameters import TrainParameters, UpdateProgressCallback
+
+from ote_cli.datasets import get_dataset_class
+from ote_cli.utils.importing import get_impl_class
+
+try:
+    import hpopt
+except ImportError as e:
+    print(e)
+    hpopt = None
+
+
+def check_hpopt_available():
+    """Check whether hpopt is avaiable"""
+
+    if hpopt is None:
+        return False
+    return True
+
+
+def get_cuda_device_count():
+    """Count number of avaiable cuda devices"""
+
+    if torch.cuda.is_available():
+        hpo_env = os.environ.copy()
+        cuda_visible_devices = hpo_env.get('CUDA_VISIBLE_DEVICES', None)
+
+        if cuda_visible_devices is None:
+            return torch.cuda.device_count()
+        else:
+            return len(cuda_visible_devices.split(','))
```

Comment:
Does torch.cuda.device_count() not work in this case?

Reply:
This is the code to use only the gpu set in the CUDA_VISIBLE_DEVICES environment variable. For example, if there are a total of 4 gpus in the system and CUDA_VISIBLE_DEVICES is set to "1,2", this code makes not to use all gpus and to use only gpus 1 and 2.

Reply:
I see, but even if there are a total of 4 gpus in the system and CUDA_VISIBLE_DEVICES is set to "1,2",
torch.cuda.device_count() returns 2. I guess this function can be removed, how do you think?

I guess you can always use just torch.cuda.device_count()

Reply:
https://github.com/pytorch/pytorch/blob/master/torch/cuda/__init__.py#L453

Reply:
I didn't know that. I removed this function. Thank you!